extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -503,23 +507,46 @@ public String getHlsUrl() throws ParsingException {
     @Override
     public List<AudioStream> getAudioStreams() throws ExtractionException {
         assertPageFetched();
-        final List<AudioStream> audioStreams = new ArrayList<>();
-        final YoutubeThrottlingDecrypter throttlingDecrypter = new YoutubeThrottlingDecrypter(getId());
 
-        try {
-            for (final Map.Entry<String, ItagItem> entry : getItags(ADAPTIVE_FORMATS,
-                    ItagItem.ItagType.AUDIO).entrySet()) {
-                final ItagItem itag = entry.getValue();
-                String url = entry.getKey();
-                url = throttlingDecrypter.apply(url);
-
-                final AudioStream audioStream = new AudioStream(url, itag);
-                if (!Stream.containSimilarStream(audioStream, audioStreams)) {
-                    audioStreams.add(audioStream);
+        if (audioStreams == null) {
```

Comment:
I don't like the fact that these three blocks of code are basically the same, but I don't know if it can be solved. Maybe the `if (streamType == StreamType.VIDEO_STREAM) {` and `if (isUrl) {` checks can be moved directly to `getItags` and `getItags` could directly return an object containing all that is needed to build a `VideoStream` or an `AudioStream`?

Reply:
Yes, I am agree with you. Whay do you think of the changes I made with https://github.com/TeamNewPipe/NewPipeExtractor/pull/663/commits/8eeb06083336d59231a602252c8b6ce5016e8cb1?